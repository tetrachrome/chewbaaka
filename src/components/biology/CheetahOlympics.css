/**
 * CheetahOlympics.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 17, 2020
 * Updated  : Jul 19, 2020
 */

div.CheetahOlympicsOuterContainer {
  margin: 0 auto;
  width: 720px;
  background-color: wheat;
  font-family: "Comic Sans MS", "Comic Sans", cursive;
  font-size: 16px;
}

div.CheetahOlympicsInnerContainer {
  text-align: center;
}

h4.CheetahOlympicsTitle {
  font-family: "Comic Sans MS", "Comic Sans", cursive;
  font-size: 24px;
  text-transform: uppercase;  
}

p.CheetahOlympicsSubtitle {
  margin-top: 40px;
  font-size: 18px;
  text-transform: uppercase;
  font-weight: 900;
}
