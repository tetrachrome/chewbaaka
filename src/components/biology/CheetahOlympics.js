/**
 * CheetahOlympics.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 17, 2020
 * Updated  : Jul 17, 2020
 */

import React from 'react'

import QuizBoard from '../shared/QuizBoard'

import QnAPopUp from '../shared/QnAPopUp'

import './CheetahOlympics.css'

export default class CheetahOlympics extends React.Component {

  static __ITEMS__ = [
    {
      title: "Lion",
      description: "80 km/hr (2nd place)",
      position: {
        top: 40,
        left: 40
      }
    },
    {
      title: "Human",
      description: "45 km/hr (7th place)",
      position: {
        top: 40,
        left: 280
      }
    },
    {
      title: "Zebra",
      description: "58 km/hr (6th place) ",
      position: {
        top: 40,
        left: 520
      }
    },
    {
      title: "Black Mamba",
      description: "32 km/hr (9th place)",
      position: {
        top: 240,
        left: 40
      }
    },
    {
      title: "Wildebeest",
      description: "78 km/hr (3rd place)",
      position: {
        top: 240,
        left: 280
      }
    },
    {
      title: "Cheetah",
      description: "110 km/hr (1st place üèÜ)",
      position: {
        top: 240,
        left: 520
      }
    },
    {
      title: "Race Horse",
      description: "60 km/h (5th place)",
      position: {
        top: 440,
        left: 40
      }
    },
    {
      title: "Elephant",
      description: "65 km/hr (4th place)",
      position: {
        top: 440,
        left: 280
      }
    },
    {
      title: "Warthog",
      description: "40 km/hr (8th place)",
      position: {
        top: 440,
        left: 520
      }
    },
    {
      title: "Jackal",
      description: "16 km/hr (10th place)",
      position: {
        top: 640,
        left: 280
      }
    },
    {
      title: "Second fastest animal in the world",
      description: "Pronghorn antelope, a North American species, sprints up to 86 km/hr.",
      position: {
        top: 840,
        left: 40
      }
    },
    {
      title: "Fastest animal in the world",
      description: "Peregrine falcon, a bird species, dives at 360km/hr.",
      fontSize: 16,
      position: {
        top: 840,
        left: 520
      }
    },
  ]

  render() {
    return (
      <div className="CheetahOlympicsOuterContainer">
        <div className="CheetahOlympicsInnerContainer">
          <h4 className="CheetahOlympicsTitle">Cheetah Olympics</h4>

          <p>Frankie Fredericks, Namibian Olympic silver medal sprinter, ran 200 meters in 19.92 seconds.</p>

          <p>The cheetah can run the same distance in 6.54 seconds.</p>

          <p className="CheetahOlympicsSubtitle">Who will win the race?</p>
          <p>Which is the fastest land animal in Africa?</p>

          <QuizBoard
            width={720}
            height={1040}
            items={CheetahOlympics.__ITEMS__}
          />
        </div>
        <div className="CheetahOlympicsQnAPopUpContainer">
          <QnAPopUp
            content="The pronghorn antelope is the fastest land animal in North America. When the first cheetahs ancestors roamed in North America, they used to prey on pronghorn antelopes."
          />
        </div>
      </div>
    );
  }
}
