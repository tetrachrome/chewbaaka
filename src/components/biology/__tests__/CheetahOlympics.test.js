/**
 * CheetahOlympics.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 19, 2020
 * Updated  : Jul 19, 2020
 */

import React from 'react'

import { render } from '@testing-library/react'

import renderer from 'react-test-renderer'

import CheetahOlympics from '../CheetahOlympics'

test('renders CheetahOlympics component', () => {
  render(
    <CheetahOlympics />
  );
});

test('CheetahOlympics component snapshot', () => {
  const tree = renderer
    .create(
      <CheetahOlympics />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
