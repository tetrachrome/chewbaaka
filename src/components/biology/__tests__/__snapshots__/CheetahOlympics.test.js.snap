// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheetahOlympics component snapshot 1`] = `
<div
  className="CheetahOlympicsOuterContainer"
>
  <div
    className="CheetahOlympicsInnerContainer"
  >
    <h4
      className="CheetahOlympicsTitle"
    >
      Cheetah Olympics
    </h4>
    <p>
      Frankie Fredericks, Namibian Olympic silver medal sprinter, ran 200 meters in 19.92 seconds.
    </p>
    <p>
      The cheetah can run the same distance in 6.54 seconds.
    </p>
    <p
      className="CheetahOlympicsSubtitle"
    >
      Who will win the race?
    </p>
    <p>
      Which is the fastest land animal in Africa?
    </p>
    <div
      className="QuizBoardOuterContainer QuizBoardOuterContainer_DebugColor DebugWireframe"
      style={
        Object {
          "height": 1040,
          "width": 720,
        }
      }
    >
      <div
        className="QuizBoardInnerContainer"
      >
        <div
          className="QuizBoardItemContainer"
          style={
            Object {
              "left": 40,
              "top": 40,
            }
          }
        >
          <div
            className="ui fade reveal"
          >
            <div
              className="ui visible content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
                style={
                  Object {
                    "backgroundColor": "#6e9ce9",
                  }
                }
              >
                <p
                  className="TextRevealWithTextCoverTitle"
                >
                  Lion
                </p>
              </div>
            </div>
            <div
              className="ui hidden content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
              >
                <p
                  className="TextRevealWithTextCoverDescription"
                >
                  80 km/hr (2nd place)
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="QuizBoardItemContainer"
          style={
            Object {
              "left": 280,
              "top": 40,
            }
          }
        >
          <div
            className="ui fade reveal"
          >
            <div
              className="ui visible content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
                style={
                  Object {
                    "backgroundColor": "#6e9ce9",
                  }
                }
              >
                <p
                  className="TextRevealWithTextCoverTitle"
                >
                  Human
                </p>
              </div>
            </div>
            <div
              className="ui hidden content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
              >
                <p
                  className="TextRevealWithTextCoverDescription"
                >
                  45 km/hr (7th place)
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="QuizBoardItemContainer"
          style={
            Object {
              "left": 520,
              "top": 40,
            }
          }
        >
          <div
            className="ui fade reveal"
          >
            <div
              className="ui visible content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
                style={
                  Object {
                    "backgroundColor": "#6e9ce9",
                  }
                }
              >
                <p
                  className="TextRevealWithTextCoverTitle"
                >
                  Zebra
                </p>
              </div>
            </div>
            <div
              className="ui hidden content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
              >
                <p
                  className="TextRevealWithTextCoverDescription"
                >
                  58 km/hr (6th place) 
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="QuizBoardItemContainer"
          style={
            Object {
              "left": 40,
              "top": 240,
            }
          }
        >
          <div
            className="ui fade reveal"
          >
            <div
              className="ui visible content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
                style={
                  Object {
                    "backgroundColor": "#6e9ce9",
                  }
                }
              >
                <p
                  className="TextRevealWithTextCoverTitle"
                >
                  Black Mamba
                </p>
              </div>
            </div>
            <div
              className="ui hidden content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
              >
                <p
                  className="TextRevealWithTextCoverDescription"
                >
                  32 km/hr (9th place)
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="QuizBoardItemContainer"
          style={
            Object {
              "left": 280,
              "top": 240,
            }
          }
        >
          <div
            className="ui fade reveal"
          >
            <div
              className="ui visible content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
                style={
                  Object {
                    "backgroundColor": "#6e9ce9",
                  }
                }
              >
                <p
                  className="TextRevealWithTextCoverTitle"
                >
                  Wildebeest
                </p>
              </div>
            </div>
            <div
              className="ui hidden content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
              >
                <p
                  className="TextRevealWithTextCoverDescription"
                >
                  78 km/hr (3rd place)
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="QuizBoardItemContainer"
          style={
            Object {
              "left": 520,
              "top": 240,
            }
          }
        >
          <div
            className="ui fade reveal"
          >
            <div
              className="ui visible content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
                style={
                  Object {
                    "backgroundColor": "#6e9ce9",
                  }
                }
              >
                <p
                  className="TextRevealWithTextCoverTitle"
                >
                  Cheetah
                </p>
              </div>
            </div>
            <div
              className="ui hidden content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
              >
                <p
                  className="TextRevealWithTextCoverDescription"
                >
                  110 km/hr (1st place üèÜ)
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="QuizBoardItemContainer"
          style={
            Object {
              "left": 40,
              "top": 440,
            }
          }
        >
          <div
            className="ui fade reveal"
          >
            <div
              className="ui visible content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
                style={
                  Object {
                    "backgroundColor": "#6e9ce9",
                  }
                }
              >
                <p
                  className="TextRevealWithTextCoverTitle"
                >
                  Race Horse
                </p>
              </div>
            </div>
            <div
              className="ui hidden content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
              >
                <p
                  className="TextRevealWithTextCoverDescription"
                >
                  60 km/h (5th place)
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="QuizBoardItemContainer"
          style={
            Object {
              "left": 280,
              "top": 440,
            }
          }
        >
          <div
            className="ui fade reveal"
          >
            <div
              className="ui visible content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
                style={
                  Object {
                    "backgroundColor": "#6e9ce9",
                  }
                }
              >
                <p
                  className="TextRevealWithTextCoverTitle"
                >
                  Elephant
                </p>
              </div>
            </div>
            <div
              className="ui hidden content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
              >
                <p
                  className="TextRevealWithTextCoverDescription"
                >
                  65 km/hr (4th place)
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="QuizBoardItemContainer"
          style={
            Object {
              "left": 520,
              "top": 440,
            }
          }
        >
          <div
            className="ui fade reveal"
          >
            <div
              className="ui visible content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
                style={
                  Object {
                    "backgroundColor": "#6e9ce9",
                  }
                }
              >
                <p
                  className="TextRevealWithTextCoverTitle"
                >
                  Warthog
                </p>
              </div>
            </div>
            <div
              className="ui hidden content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
              >
                <p
                  className="TextRevealWithTextCoverDescription"
                >
                  40 km/hr (8th place)
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="QuizBoardItemContainer"
          style={
            Object {
              "left": 280,
              "top": 640,
            }
          }
        >
          <div
            className="ui fade reveal"
          >
            <div
              className="ui visible content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
                style={
                  Object {
                    "backgroundColor": "#6e9ce9",
                  }
                }
              >
                <p
                  className="TextRevealWithTextCoverTitle"
                >
                  Jackal
                </p>
              </div>
            </div>
            <div
              className="ui hidden content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
              >
                <p
                  className="TextRevealWithTextCoverDescription"
                >
                  16 km/hr (10th place)
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="QuizBoardItemContainer"
          style={
            Object {
              "left": 40,
              "top": 840,
            }
          }
        >
          <div
            className="ui fade reveal"
          >
            <div
              className="ui visible content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
                style={
                  Object {
                    "backgroundColor": "#6e9ce9",
                  }
                }
              >
                <p
                  className="TextRevealWithTextCoverTitle"
                >
                  Second fastest animal in the world
                </p>
              </div>
            </div>
            <div
              className="ui hidden content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
              >
                <p
                  className="TextRevealWithTextCoverDescription"
                >
                  Pronghorn antelope, a North American species, sprints up to 86 km/hr.
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="QuizBoardItemContainer"
          style={
            Object {
              "left": 520,
              "top": 840,
            }
          }
        >
          <div
            className="ui fade reveal"
          >
            <div
              className="ui visible content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
                style={
                  Object {
                    "backgroundColor": "#6e9ce9",
                  }
                }
              >
                <p
                  className="TextRevealWithTextCoverTitle"
                >
                  Fastest animal in the world
                </p>
              </div>
            </div>
            <div
              className="ui hidden content"
            >
              <div
                className="TextRevealWithTextCoverOuterContainer"
              >
                <p
                  className="TextRevealWithTextCoverDescription"
                >
                  Peregrine falcon, a bird species, dives at 360km/hr.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="QnAPopUpContainer QnAPopUpContainer_DebugColor DebugWireframe"
    >
      <div
        className="QnAPopUpButtonGroupContainer QnAPopUpButtonGroupContainer_DebugColor DebugWireframe"
      >
        <div
          className="ui blue large label"
          onClick={[Function]}
        >
          Did You Know
          <button
            className="ui orange circular icon button QuestionButton"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="question icon"
              onClick={[Function]}
            />
          </button>
        </div>
      </div>
      <div
        className="QnAPopUpTextSecondaryBubbleContainer QnAPopUpTextSecondaryBubbleContainer_DebugColor DebugWireframe"
      >
        <div
          className="TextBubbleOuterContainer"
          hidden={true}
          style={
            Object {
              "backgroundColor": "rgb(undefined)",
              "height": 45,
              "width": 45,
            }
          }
        >
          <div
            className="TextBubbleInnerContainer"
          >
            <div
              className="TextBubbleInnerCore"
            >
              <h1
                style={
                  Object {
                    "color": "rgb(undefined)",
                    "fontSize": 3.6,
                  }
                }
              >
                
              </h1>
              <p
                style={
                  Object {
                    "color": "rgb(undefined)",
                    "fontSize": 2.25,
                  }
                }
              >
                
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="QnAPopUpMainTextBubbleContainer QnAPopUpMainTextBubbleContainer_DebugColor DebugWireframe"
      >
        <div
          className="TextBubbleOuterContainer"
          hidden={true}
          style={
            Object {
              "backgroundColor": "rgb(undefined)",
              "height": 256,
              "width": 256,
            }
          }
        >
          <div
            className="TextBubbleInnerContainer"
          >
            <div
              className="TextBubbleInnerCore"
            >
              <h1
                style={
                  Object {
                    "color": "rgb(undefined)",
                    "fontSize": 20.48,
                  }
                }
              >
                Did you know?
              </h1>
              <p
                style={
                  Object {
                    "color": "rgb(undefined)",
                    "fontSize": 12.8,
                  }
                }
              >
                The pronghorn antelope is the fastest land animal in North America. When the first cheetahs ancestors roamed on the continent, they used to prey on pronghorn antelopes.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
